src/consumer/runner.js

```diff
@@ -181,8 +181,8 @@ module.exports = class Runner extends EventEmitter {
           })
         }
 
-        // In case of errors, commit the previously consumed offsets
-        await this.consumerGroup.commitOffsets()
+        // In case of errors, commit the previously consumed offsets if needed
+        await this.consumerGroup.commitOffsetsIfNecessary()
```

Comment:
I am not sure if this one should be `commitOffsetsIfNecessary` as in case if we get an error we want to commit offsest always (except if we have `this.autoCommit` false) but `commitOffsetsIfNecessary` also checks last commit time which means that we may not commit all processed messages in case if previous commit was too near...

Reply:
I see. You are right. Maybe we need a third commitOffsetIfEnabled function?

Reply:
There are 2 functions available on runner which basically already checking all the things properly (not sure why we don't just use them...) I am wondering if we can rename those to more appropriate names and use those or just move `autoCommit` check logic in to `consumerGroup.commitOffsets` and `consumerGroup.commitOffsetsIfNecessary` functions and delete below one completely, this should make code cleaner and simpler  @Nevon what do you think ?

```js
 autoCommitOffsets() {
    if (this.autoCommit) {
      return this.consumerGroup.commitOffsets()
    }
  }

  autoCommitOffsetsIfNecessary() {
    if (this.autoCommit) {
      return this.consumerGroup.commitOffsetsIfNecessary()
    }
  }
```

Reply:
![Gif of Tina Fey saying "Well, here we are 32 years later"](https://media.giphy.com/media/aHnPa5VYzfQhq/giphy.gif)

`consumerGroup.commitOffsets` is exposed to the `eachBatch` interface as well as via `consumer.commitOffsets` as a way to manually commit offsets, so that really shouldn't really need to know about `autoCommit`. I think the offending line (185) should just be changed to calling  `autoCommitOffsetsIfNecessary` and then the bug will be fixed.